scripting/src/main/java/org/infinispan/scripting/impl/ScriptMetadataParser.java

```diff
@@ -0,0 +1,102 @@
+package org.infinispan.scripting.impl;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.infinispan.scripting.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+public class ScriptMetadataParser {
+   private static final Log log = LogFactory.getLog(ScriptMetadataParser.class, Log.class);
+   private static final String DEFAULT_SCRIPT_EXTENSION = "js";
+   private static final Pattern METADATA_COMMENT = Pattern.compile("^(?:#|//|;;)\\s*(.+)");
+   private static final Pattern METADATA_KV = Pattern
+         .compile("\\s*(\\w+)\\s*=\\s*((?:\")[^\"]*(?:\")|\\[[\\w,\\s]*\\]|[^,=\\s\"]+)\\s*,?");
```

Comment:
You read through the regex ? I'm impressed :)
I guess, since I'm unquoting the value in line 41, I don't need the non-capturing group.